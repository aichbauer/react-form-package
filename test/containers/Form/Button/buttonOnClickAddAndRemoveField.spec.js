import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';

import {
  Form,
  Field,
  Button,
} from '../../../../src';

configure({ adapter: new Adapter() });

test('Render Form with button | add and remove fields | click button', () => {
  const onClickMockRemoveField = jest.fn();
  const onClickMockAddField = jest.fn();

  const companies = [
    {
      id: 'company-0',
    },
  ];

  const myComponent = (
    <Form>
      {companies.map((company) => (
        <div>
          <Field
            id={company.id}
            placeholder={`Company ${company.id.split('-')[1]}`}
            type="text"
            required
          />
          <Button
            id="removeField"
            rfpRole="removeField"
            type="button"
            fieldId={company.id}
            onClick={onClickMockRemoveField}
          >
            Remove Company
          </Button>
        </div>
      ))
      }
      <div>
        <Button
          id="addField"
          rfpRole="addField"
          type="button"
          field={{
            id: 'company-1',
            type: 'text',
            required: true,
          }}
          onClick={onClickMockAddField}
        >
          Add Company
        </Button>
      </div>
    </Form>
  );

  const tree = renderer.create(myComponent).toJSON();
  const wrapper = mount(myComponent);
  let state = wrapper.state();

  expect(Object.keys(state.data).length).toBe(1);

  wrapper.find('#addField').at(0).simulate('click');
  wrapper.update();

  state = wrapper.state();
  expect(Object.keys(state.data).length).toBe(2);

  wrapper.find('#removeField').at(0).simulate('click');
  wrapper.update();

  state = wrapper.state();
  expect(Object.keys(state.data).length).toBe(1);

  expect(onClickMockRemoveField).toHaveBeenCalled();
  expect(onClickMockAddField).toHaveBeenCalled();
  expect(tree).toMatchSnapshot();
});
