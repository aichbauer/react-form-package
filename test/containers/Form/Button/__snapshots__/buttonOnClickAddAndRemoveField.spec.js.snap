// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Form with button | add and remove fields | click button 1`] = `
<form>
  <div>
    <input
      className="rfp-input"
      id="company-0"
      invalid={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Company 0"
      required={true}
      type="text"
      valid={false}
      value=""
    />
    <button
      className="rfp-button"
      disabled={false}
      fieldId="company-0"
      id="removeField"
      onClick={[Function]}
      rfpRole="removeField"
      type="button"
    >
      Remove Company
    </button>
  </div>
  <div>
    <button
      className="rfp-button"
      disabled={false}
      field={
        Object {
          "id": "company-1",
          "required": true,
          "type": "text",
        }
      }
      id="addField"
      onClick={[Function]}
      rfpRole="addField"
      type="button"
    >
      Add Company
    </button>
  </div>
</form>
`;
